### PR TITLE
chore: revamp input manipulation unit tests

### PR DESCRIPTION
**Pull request by taharallouche**
## Description
The changes only concern rewriting the unit tests for input manipulation utils, namely:
- Used `np.testing.assert_equal_array` for array comparisons.
- Used pytest parametrization for running multiple test cases on the same function behaviour.
- Used pytest raises's `match` argument to assert equal expected error message.

No existing tests were removed/modified.
 
This a small PR suggestion whose purpose is mainly to get myself used to the dev process on `fairlearn` before potentially tackling more useful issues.

## Tests

- [x] no new tests required
- [x] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [] user guide added or updated
- [x] API docs added or updated
- [x] example notebook added or updated

### CODE REVIEW

File: test/unit/utils/test_input_manipulations.py
Comment (adrinjalali) [inline test/unit/utils/test_input_manipulations.py:21]: you can move import statements to the top of the file

**Review by adrinjalali — COMMENTED** [review]
Otherwise LGTM. Thanks @taharallouche 

I'll let @TamaraAtanasoska have a look as a second reviewer

**Review by TamaraAtanasoska — APPROVED** [review]
LGTM! Thanks for your tests refactor Tahar! My only comment is about the PR description - all the change options are ticked. In this case I would tick `"existing tests adjusted"` and none for the documentation part. 

Btw, have you joined our Discord server @taharallouche? I see there are are a few other PRs for review, you might find answers to questions and support in the #development channel on here: https://discord.com/invite/R22yCfgsRn

**Comment by taharallouche** [comment]
> LGTM! Thanks for your tests refactor Tahar! My only comment is about the PR description - all the change options are ticked. In this case I would tick `"existing tests adjusted"` and none for the documentation part.

Hey @TamaraAtanasoska sorry about that, I've paid more attention to the boxes to check in the next PRs :sweat_smile: 

> Btw, have you joined our Discord server @taharallouche? I see there are are a few other PRs for review, you might find answers to questions and support in the #development channel on here: https://discord.com/invite/R22yCfgsRn

I just joined the server thanks for the invitation :pray:

**Pull request merged**
